skbio/metadata/tests/test_interval.py

```diff
@@ -446,6 +446,34 @@ def test_init_upper_bound_lt_lower_bound(self):
         with self.assertRaises(ValueError):
             IntervalMetadata(-1)
 
+    def test_upper_bound_is_none(self):
```

Comment:
`__eq__` should continue to check `upper_bound` because bounded vs unbounded interval metadata objects shouldn't compare equal.

You can define an *expected* `IntervalMetadata` object that has the expected `upper_bound` (in addition to whatever other state is expected), and use `self.assertEqual` to compare observed to expected. Is there a reason this strategy won't work? This is a really common pattern in unit testing.